pymeasure/instruments/oxfordinstruments/base.py

```diff
@@ -38,25 +38,47 @@ class OxfordVISAError(Exception):
     pass
 
 
-class OxfordInstrumentsAdapter(VISAAdapter):
-    """Adapter class for the VISA library using PyVISA to communicate
-    with instruments.
+class OxfordInstrumentsBase(Instrument):
+    """Base instrument for devices from Oxford Instruments.
+
     Checks the replies from instruments for validity.
 
-    :param resource_name: VISA resource name that identifies the address
+    :param adapter: A string, integer, or :py:class:`~pymeasure.adapters.Adapter` subclass object
+    :param string name: The name of the instrument. Often the model designation by default.
     :param max_attempts: Integer that sets how many attempts at getting a
         valid response to a query can be made
-    :param kwargs: key-word arguments for constructing a PyVISA Adapter
+    :param \\**kwargs: In case ``adapter`` is a string or integer, additional arguments passed on
+        to :py:class:`~pymeasure.adapters.VISAAdapter` (check there for details).
+        Discarded otherwise.
     """
 
     timeoutError = VisaIOError(-1073807339)
 
     regex_pattern = r"^([a-zA-Z])[\d.+-]*$"
 
-    def __init__(self, resource_name, max_attempts=5, **kwargs):
-        super().__init__(resource_name, **kwargs)
+    def __init__(self, adapter, name="OxfordInstruments Base", max_attempts=5, **kwargs):
+        kwargs.setdefault('read_termination', '\r')
+
+        super().__init__(adapter,
+                         name=name,
+                         includeSCPI=False,
+                         asrl={
+                             'baud_rate': 9600,
+                             'data_bits': 8,
+                             'parity': 0,
+                             'stop_bits': 20,
```

Comment:
```suggestion
                             'parity': vconst.Parity.none,
                             'stop_bits': vconst.StopBits.two,
```
at least for me this is necessary, I am going to check with the latest version of `visa`

Reply:
newest version of pyvisa and visa, and this works, no idea how it could have worked without it

Reply:
Because these constants have the following values (in my pyvisa installation):
~~~
vconst.Parity.none = 0
vconst.StopBits.two = 20
~~~

Reply:
yes, I see that it was so, I am not sure how to deal with that. I am using

PyVISA==1.12.0
visa==1.0.0

But I am not 100% sure whether I am now using the NI visa library or the Keysight one, and even so I am not sure whether that would make a difference here

Reply:
Ok, so, I am confused now. With clean environments that only contain what is needed I can run it with the numbers instead of the `vconst.Parity.none` things. With the production environment which my setup is running on, it does not work -- so this is related to my system, not to visa versions. With the visa constants, it works in all cases for me. Not sure what would be better to keep

Reply:
We need the visa system anyways, so I do not see any problem using the constants. They are better to read anyway (none compared to 0 and two compared to 20).
If the constants work with any backend, it is one reason more, to use them.